cloud/scope/machine.go

```diff
@@ -93,9 +93,9 @@ func (m *MachineScope) Cloud() cloud.Cloud {
 // Zone returns the FailureDomain for the GCPMachine.
 func (m *MachineScope) Zone() string {
 	if m.Machine.Spec.FailureDomain == nil {
-		return ""
+		fd := m.ClusterGetter.FailureDomains().GetIDs()
+		return *fd[0]
```

Comment:
I got something like `panic: runtime error: index out of range [0] with length 0` looks like the slice is empty
So with the hard coded return value like `return "us-central1-c"` I managed to pass the tests